### PR TITLE
Naming and refactoring of Coordinator, Memory and UI components

### PR DESCRIPTION
**Pull request by philippjfr**
Renames:

- `Coordinator` (new base class)
- `Assistant` -> `DependencyResolver`: First implementation of an `Coordinator`
- `PlanningAssistant` -> `Planner`: Second implementation of an `Coordinator`

- `UI` (new base class for high-level UI)
- `LumenAI` -> `ChatUI`
- `LumenAIExplorer` -> `ExplorerUI`

Also I'm still in the process of factoring the Assistant controls out and re-adding them at the UI level.

### CODE REVIEW

**Comment by ahuang11** [comment]
My gut is leaning towards Coordinator over Orchestrator but no strong feelings. Resolver wasn't intuitive to me when I first saw it until I read the docstring--also no strong feelings.

All the other names makes sense to me.

**Comment by philippjfr** [comment]
Sounds good, happy with `Coordinator` instead. I think I'll probably stick with `Planner` but you're right that `Resolver` isn't great.

**Comment by ahuang11** [comment]
Planning to merge this and start the prompt consolidation refactor soon. We can update the resolver name later if we can think of a better name.

File: lumen/ai/ui.py [hunk 1]
```diff
@@ -18,61 +18,50 @@
...
     """
 
     analyses = param.List(default=[], doc="""
         List of custom analyses. If provided the AnalysesAgent will be added."""
     )
 
-    assistant = param.ClassSelector(
-        class_=Assistant, default=PlanningAssistant, is_instance=False, doc="""
-        The Assistant class that will be responsible for coordinating the Agents."""
+    orchestrator = param.ClassSelector(
```
Comment (ahuang11) [inline lumen/ai/ui.py:41]:
```suggestion
    coordinator = param.ClassSelector(
```

File: lumen/ai/coordinator.py [hunk 2]
```diff
@@ -50,41 +48,40 @@ class AgentChainLink(param.Parameterized):
...
+    step along the graph.
     """
 
     agents = param.List(default=[ChatAgent], doc="""
-        List of agents the assistant will coordinate.""")
+        List of agents to coordinate.""")
 
     demo_inputs = param.List(default=DEMO_MESSAGES, doc="""
-        List of instructions to demo the capabilities of the assistant.""")
+        List of instructions to demo the orchestrator.""")
```
Comment (ahuang11) [inline lumen/ai/coordinator.py:63]:
```suggestion
        List of instructions to demo the coordinator.""")
```

**Comment by ahuang11** [comment]
What about `Solver`? Or `DependencySolver` or `DependencyResolver` instead of `Resolver`

**Review by ahuang11 — APPROVED** [review]

**Comment by ahuang11** [comment]
Merging to start refactoring the prompts; we can change the names later if we decide on something.

**Pull request merged**